.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,13 @@
+name: Pre-commit
+
```

Comment:
This workflow, in effect, replaces our style_checks workflow.  The style checks workflow had been running the 'Prettier' npm package directly, and ensuring that PRs would meet prettier standards.

This is now using a pre-commit based workflow.  Pre-commit is a tool that manages git hooks in a pretty convenient way.  This workflow runs out pre-commit hooks against all files, and will pass/fail accordingly.  Since we have added the prettier pre-commit hook (see the precommit config yaml also in this PR), this workflow will check for Prettier compliance.  So, there's no change in prettier functionality.

Why use pre-commit?  Pre-commit will prevent developers from commiting code that does not meet the standards checked by the hooks.  
Why run a workflow if code is checked before being commited?  Pre-commit will only run if a developer has run 'pre-commit install' inside their project.  Since we can only document and encourage devs to install pre-commit and cannot enforce its use, we will check for compliance as a required PR status check